File: 01/main.py
import sys

nums = []
for line in sys.stdin:
    nums.append(int(line.rstrip()))

for i in range(0, len(nums)):
    for j in range(i+1, len(nums)):
        if nums[i] + nums[j] == 2020:
            print("PART 1: {}".format(nums[i] * nums[j]))
        for k in range(j+1, len(nums)):
            if nums[i] + nums[j] + nums[k]== 2020:
                print("PART 2: {}".format(nums[i] * nums[j] * nums[k]))

File: 01/main.php
<?php

$nums = [];

while (($line = fgets(STDIN)) !== false) {
    if ($line !== false) {
        if ($line === "\n") {
            continue;
        }
        $num = rtrim($line);
        $nums[] = $num;
    }
}

for ($i = 0; $i < count($nums); $i++) {
    for ($j = $i+1; $j < count($nums); $j++) {
        if ($nums[$i] + $nums[$j] == 2020 ) {
            printf("PART 1: %d\n", $nums[$i] * $nums[$j]);
        }
        for ($k = $j+1; $k < count($nums); $k++) {
            if ($nums[$i] + $nums[$j] + $nums[$k] == 2020) {
                printf("PART 2: %d\n", $nums[$i] * $nums[$j] * $nums[$k]);
            }
        }
    }
}
